### DISCUSSION TITLE
isInline() started to fail since v2.10.27

### DISCUSSION THREAD

**Discussion by Nielsvanpach** [filamentphp/filament#1914] [Help]
### Package

filament/forms

### Package Version

v2.10.27

### Laravel Version

v8.83.4

### PHP Version

PHP 8.1

### Bug description

After upgrading to v2.10.27 we got some failing code when checking if a field is inline with `isInline()`. I assume it might be related to https://github.com/laravel-filament/filament/pull/1883.

### Steps to reproduce

```php
$toggle = Toggle::make('is_admin')->inline();

$toggle->isInline();
```

or

```php
$toggle = Checkbox::make('is_admin');

$toggle->isInline();
```

```
Error : Typed property Filament\Forms\Components\Component::$container must not be accessed before initialization
```

### Relevant log output

```shell
Error : Typed property Filament\Forms\Components\Component::$container must not be accessed before initialization
 /Users/niels/dev/spatie/***/vendor/filament/forms/src/Components/Concerns/BelongsToContainer.php:21
 /Users/niels/dev/spatie/***/vendor/filament/forms/src/Components/Concerns/HasInlineLabel.php:20
 /Users/niels/dev/spatie/***/vendor/filament/forms/src/Components/Concerns/CanBeInline.php:20
 /Users/niels/dev/spatie/***.php:120
```

**Comment by danharrin** [comment]
Yeah, why are you calling `isInline()` like that?

**Comment by ryangjchandler** [comment]
Isn't the problem here that the field is being instantiated on it's own, outside of a `ComponentContainer`?

**Comment by danharrin** [comment]
Yeah, this would have broken pre-v2.10.27 if a closure was used inside `inline()`.

**Comment by danharrin** [comment]
I'm just interested to see why `isInline()` was being called directly in the first place, we can recommend a more robust solution.

**Comment by Nielsvanpach** [comment]
> I'm just interested to see why isInline() was being called directly in the first place, we can recommend a more robust solution.

We are dynamically generating Filament forms. `isInline()` is one of the checks used in our tests to verify if Field methods like `inline()` are working correctly. 

What solution do you propose? Specifically defining a container solves the issue in the tests, but I though it was worth mentioning this here as a public Field method broke in a patch release.

@ryangjchandler examples are coming from the docs.
https://filamentphp.com/docs/2.x/forms/fields#checkbox
https://filamentphp.com/docs/2.x/forms/fields#toggle

**Comment by ryangjchandler** [comment]
The public method broke in a patch release, but tbh it shouldn't have been working before anyway.

Those methods need to be marked as public for us to interact with them correctly, but should likely be marked as `@internal` and used at your own risk since we don't document them as part of the public API for the forms package. 

All fields need to be bound to a `ComponentContainer`, so I'd recommend taking this approach in your test. You can create a fake Livewire component for the container too.

I'm going to mark this as a `wontfix` and move it to a discussion as well.

**Reply by ryangjchandler** [reply]
Dan and I have discussed testing quite a bit as well. You can see how we do some of our testing internally here: https://github.com/laravel-filament/filament/blob/7461b055e856c7c5e794bebe5eef10d80054c689/tests/src/Forms/HiddenComponentsTest.php#L11

Are you testing an entire form component (Livewire component with a form), or an individual field?

**Reply by danharrin** [reply]
Yeah, or you could maybe try something like `Livewire::test()->instance()->form->getComponent('is_admin')`, then you dont even need to set up a container as Livewire will do that for you.
